[TORCH] Add support for fix Op

sharavana20 opened the pull request:
- Added the support for fix Op
- Since this is an alias for torch.trunc
- Added the e2e test

This implementation addresses and closes https://github.com/llvm/torch-mlir/issues/4193

sharavana20 commented:
@vivekkhandelwal1 I don’t have permission to assign reviewers directly , but could you please take a look at this PR. Your review would be much appreciated!

zjgarvey reviewed (approved):
LGTM. Out of curiosity, have you tried adding this op to the fx decomposition list for import?

sharavana20 commented:
> LGTM. Out of curiosity, have you tried adding this op to the fx decomposition list for import?

Thanks for the quick response @zjgarvey. I tried only using fx.export_and_import method without decomposition list

sharavana20 commented on lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:
```diff
@@ -8780,6 +8780,20 @@ class DecomposeAtenTruncOp : public OpRewritePattern<AtenTruncOp> {
+                                PatternRewriter &rewriter) const override {
+    Value self = op.getSelf();
+    auto resultTy = dyn_cast<ValueTensorType>(op.getType());
+    rewriter.replaceOpWithNewOp<AtenTruncOp>(op, resultTy, self);
```
Once again, thank you for the quick review, @zjgarvey! I've removed the unnecessary dyn_cast and updated the pattern to use op.getType() directly. The change has been pushed.

sharavana20 commented:
Once again, thank you for the quick review, @zjgarvey! I've removed the unnecessary dyn_cast and updated the pattern to use op.getType() directly. The change has been pushed.
